bindings/influx/influx.go

```diff
@@ -95,31 +114,60 @@ func (i *Influx) getInfluxMetadata(metadata bindings.Metadata) (*influxMetadata,
 
 // Operations returns supported operations.
 func (i *Influx) Operations() []bindings.OperationKind {
-	return []bindings.OperationKind{bindings.CreateOperation}
+	return []bindings.OperationKind{bindings.CreateOperation, queryOperation}
 }
 
 // Invoke called on supported operations.
 func (i *Influx) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
-	var jsonPoint map[string]interface{}
-	err := json.Unmarshal(req.Data, &jsonPoint)
-	if err != nil {
-		return nil, errors.New("Influx Error: Cannot convert request data")
+	switch req.Operation {
+	case bindings.CreateOperation:
+		var jsonPoint map[string]interface{}
+		err := json.Unmarshal(req.Data, &jsonPoint)
+		if err != nil {
+			return nil, ErrInvalidRequestData
+		}
+
+		line := fmt.Sprintf("%s,%s %s", jsonPoint["measurement"], jsonPoint["tags"], jsonPoint["values"])
+
+		// write the point
+		err = i.writeAPI.WriteRecord(context.Background(), line)
+		if err != nil {
+			return nil, ErrCannotWriteRecord
+		}
+		return nil, nil
+	case queryOperation:
+		if req.Metadata == nil {
+			return nil, ErrMetadataMissing
+		}
+		i.logger.Debugf("operation: %v", req.Operation)
+
+		s, ok := req.Metadata[rawQueryKey]
+		if !ok || s == "" {
+			return nil, ErrMetadataRawNotFound
+		}
+
+		res, err := i.queryAPI.QueryRaw(context.Background(), s, influxdb2.DefaultDialect())
+		if err != nil {
+			return nil, errors.Wrap(err, "do query influx err")
+		}
+
+		resp := &bindings.InvokeResponse{
+			Data: []byte(res),
+			Metadata: map[string]string{
+				respOperatorKey: string(req.Operation),
+				rawQueryKey:     s,
+			},
+		}
+		return resp, nil
+	default:
+		i.logger.Warnf("unsupported operation invoked")
+		return nil, ErrInvalidRequestOperation
 	}
-
-	line := fmt.Sprintf("%s,%s %s", jsonPoint["measurement"], jsonPoint["tags"], jsonPoint["values"])
-
-	// write the point
-	err = i.writeAPI.WriteRecord(context.Background(), line)
```

Comment:
Why remove this?

Reply:
I didn't remove the function there, I just moved it to `case bindings.CreateOperation`